### PR TITLE
Add a link to the Google Code Review Guidelines in the PR template

### PR DESCRIPTION
**Pull request by matias-pavez-dx**
### Description

- Add a link to the Google Reviewing guidelines, including quick access links for reviewers and authors.
- Clarify the scope of DeepX guidelines.

### Motivation and Context
Google Code Review guidelines (https://google.github.io/eng-practices/) are well-written and cover critical aspects for authors and reviewers in detail.

### How Has This Been Tested?

Markdown viewer.

<br>
---

- [PR Guidelines](https://github.com/DeepX-inc/.github/blob/main/pull_request_guidelines.md)

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -16,4 +16,5 @@
 <br>
 ---
 
-- [PR Guidelines](https://github.com/DeepX-inc/.github/blob/main/pull_request_guidelines.md)
+- [DeepX PR Guidelines](https://github.com/DeepX-inc/.github/blob/main/pull_request_guidelines.md) - for descriptions and metadata.
+- [Google Code Review Guidelines](https://google.github.io/eng-practices/) - for [authors](https://google.github.io/eng-practices/review/reviewer/) and [reviewers](https://google.github.io/eng-practices/review/reviewer/).
```
Comment (matias-pavez-dx) [inline .github/pull_request_template.md]:
```suggestion
- [Google Code Review Guidelines](https://google.github.io/eng-practices/) - for [authors](https://google.github.io/eng-practices/review/developer/) and [reviewers](https://google.github.io/eng-practices/review/reviewer/).
```

**Review by DmitryNeverov-dx — APPROVED** [review]
LGTM
Shall we also start calling PRs CLs? 🙂 
Maybe that superficial change could be a chance to reiterate good review practices.

**Comment by Mattia-DeepX** [comment]
I would like to add semantic reviewing guidelines to this.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -16,4 +16,5 @@
 <br>
 ---
 
-- [PR Guidelines](https://github.com/DeepX-inc/.github/blob/main/pull_request_guidelines.md)
+- [DeepX PR Guidelines](https://github.com/DeepX-inc/.github/blob/main/pull_request_guidelines.md) - for descriptions and metadata.
+- [Google Code Review Guidelines](https://google.github.io/eng-practices/) - for [authors](https://google.github.io/eng-practices/review/developer/) and [reviewers](https://google.github.io/eng-practices/review/reviewer/).
```
Comment (matias-pavez-dx) [inline .github/pull_request_template.md:20]:
```suggestion
- [DeepX PR Guidelines](https://github.com/DeepX-inc/.github/blob/main/pull_request_guidelines.md) - for descriptions and metadata.
- [Google Code Review Guidelines](https://google.github.io/eng-practices/) - for [authors](https://google.github.io/eng-practices/review/developer/) and [reviewers](https://google.github.io/eng-practices/review/reviewer/).
- [Semantic Code Reviews](https://www.m31coding.com/blog/semantic-reviews.html) - Simple and direct comments without drama.
```

**Comment by matias-pavez-dx** [comment]
@Mattia-DeepX 
> I would like to add semantic reviewing guidelines to this.

Added in 390eee81f313d9ab65cfb9c5ca37a13574b203cf

@DmitryNeverov-dx 
> Shall we also start calling PRs CLs? 🙂
> Maybe that superficial change could be a chance to reiterate good review practices.

Don't know how it can help. But I feel it would create a lot of confusion.
Anyway, if that happens we can update the PR template.

**Pull request merged**